src/main/api/2/studio-api-2.yaml

```diff
@@ -1115,6 +1119,159 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /cluster:
+    get:
+      tags:
+        - cluster
+      summary: Get all the members of the cluster
+      description: 'Required Permission: "READ_CLUSTER"'
+      operationId: getClusterMembers
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  result:
+                    type: object
+                    properties:
+                      response:
+                        $ref: '#/components/schemas/ApiResponse'
+                      entities:
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/ClusterMember'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+    post:
+      tags:
+        - cluster
+      summary: Add member(s) to the cluster
+      description: 'Required Permission: "CREATE_CLUSTER"'
+      operationId: addClusterMembers
+      requestBody:
+        description: List of cluster members to add
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                clusterMembers:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/ClusterMember'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  result:
+                    type: object
+                    properties:
+                      response:
+                        $ref: '#/components/schemas/ApiResponse'
+                      entities:
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/ClusterMember'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+    patch:
+      tags:
+        - cluster
+      summary: Update (upsert) cluster members
+      description: 'Required Permission: "UPDATE_CLUSTER"'
+      operationId: updateClusterMembers
+      requestBody:
+        description: Cluster members to update (upsert)
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                clusterMembers:
```

Comment:
We're not wrapping currently the request objects. I would say remove `clusterMembers`